rdtools/system_analysis.py

```diff
@@ -0,0 +1,585 @@
+            if temperature is not None:
+                temperature = normalization.interpolate(temperature, interp_freq, max_timedelta)
+            if clearsky_poa is not None:
+                clearsky_poa = normalization.interpolate(clearsky_poa, interp_freq, max_timedelta)
+            if clearsky_temperature is not None:
+                clearsky_temperature = normalization.interpolate(clearsky_temperature, interp_freq, max_timedelta)
+            if isinstance(pv_azimuth, (pd.Series, pd.DataFrame)):
+                pv_azimuth = normalization.interpolate(pv_azimuth, interp_freq, max_timedelta)
+            if isinstance(pv_tilt, (pd.Series, pd.DataFrame)):
+                pv_tilt = normalization.interpolate(pv_tilt, interp_freq, max_timedelta)
+
+        if pv_input == 'power':
+            self.pv_power = pv
+            self.pv_energy = normalization.energy_from_power(pv, max_timedelta=max_timedelta)
+        elif pv_input == 'energy':
+            self.pv_power = None
+            self.pv_energy = pv
+
+        if temperature_input == 'cell':
+            self.cell_temperature = temperature
+            self.ambient_temperature = None
+        elif temperature_input == 'ambient':
+            self.cell_temperature = None
+            self.ambient_temperature = temperature
+
+        if clearsky_temperature_input == 'cell':
+            self.clearsky_cell_temperature = clearsky_temperature
+            self.clearsky_ambient_temperature = None
+        elif clearsky_temperature_input == 'ambient':
+            self.clearsky_cell_temperature = None
+            self.clearsky_ambient_temperature = clearsky_temperature
+
+        self.poa = poa
+        self.temperature_coefficient = temperature_coefficient
+        self.aggregation_freq = aggregation_freq
+        self.pvlib_location = pvlib_location
+        self.clearsky_poa = clearsky_poa
+        self.windspeed = windspeed
+        self.albedo = albedo
+        self.temperature_model = temperature_model
+        self.pv_azimuth = pv_azimuth
+        self.pv_tilt = pv_tilt
+        self.pv_nameplate = pv_nameplate
+        self.results = {}
+
+        # Initialize to use default filter parameters
+        self.filter_params = {
+            'normalized_filter': {},
+            'poa_filter': {},
+            'tcell_filter': {},
+            'clip_filter': {},
+            'csi_filter': {},
+            'ad_hoc_filter': None  # use this to include an explict filter
+        }
+
+    def calc_clearsky_poa(self, times=None, rescale=True, model='isotropic', **kwargs):
+        '''
+        Calculate clearsky plane-of-array irradiance and store in self.clearsky_poa
+
+        Parameters
+        ----------
+        times: Pandas DateTimeIndex for which to calculate clearsky poa
+        rescale: Whether to attempt to rescale clearsky irradiance to measured (bool)
+        model: Model for pvlib.irradiance.get_total_irradiance() (str)
```

Comment:
Python would throw a `TypeError: calc_clearsky_poa() got multiple values for argument 'model'` if you specified it twice, so no possibility of an issue there.  Though I think it is redundant to include the model in the signature, since `get_total_irradiance` already defaults to `isotropic` anyway.  Removing it from this signature would still let it get passed in through `kwargs` without any change in the user code.

Reply:
I think my past self was wanting to encode our preferred default behavior, but it does seem confusing, so I can remove it.